src/services/services.js

```diff
@@ -1,42 +1,98 @@
+    if (eventHandlers) initEventHandlers(infra, name, eventHandlers);
+  }
 };
 
-/** @type initCommands */
+/** @type ServiceFuncs['initCommands'] */
 const initCommands = (infra, serviceName, commands) => {
   /** @type Service['commands'] */
   const initialized = {};
 
-  for (const [name, fn] of Object.entries(commands)) {
-    initialized[name] = fn.bind(null, infra);
+  for (const [commandName, fn] of Object.entries(commands)) {
+    initialized[commandName] = wrapCommand(infra, fn, {
+      logPrefix: `${serviceName}/${commandName}`,
+    });
   }
 
   infra.bus.registerService(serviceName, initialized);
 };
 
-/** @type initEventHandlers */
-const initEventHandlers = (infra, handlers) => {
+/** @type ServiceFuncs['initEventHandlers'] */
+const initEventHandlers = (infra, serviceName, handlers) => {
   for (const [eventName, handler] of Object.entries(handlers)) {
-    infra.bus.subscribe(eventName, handler.bind(null, infra));
+    const wrapped = wrapEventHandler(infra, handler, {
+      logPrefix: `${serviceName}/${eventName}`,
+    });
+    infra.bus.subscribe(eventName, wrapped);
   }
 };
 
-/** @type init */
-export const init = async (infra) => {
-  for (const [name, service] of Object.entries(services)) {
-    const { commands, eventHandlers } = service;
-    if (commands) initCommands(infra, name, commands);
-    if (eventHandlers) initEventHandlers(infra, eventHandlers);
+/** @type ServiceFuncs['wrapCommand'] */
+const wrapCommand = (infra, func, options) => async (payload) => {
+  const { logger } = infra;
+  const { logPrefix } = options;
+  const { operationId } = payload.meta;
+  try {
+    const wrappedInfra = wrapInfra(infra, operationId);
+    const result = await func(wrappedInfra, payload);
+    return [null, result];
+  } catch (error) {
+    const serviceError = processServiceError(error, logger, {
+      operationId,
+      logPrefix,
+    });
+    return [serviceError, null];
   }
 };
+
+/** @type ServiceFuncs['wrapEventHandler'] */
+const wrapEventHandler = (infra, func, options) => async (payload) => {
+  const { logger } = infra;
+  const { logPrefix } = options;
+  const { operationId } = payload.meta;
+  try {
+    const wrappedInfra = wrapInfra(infra, operationId);
+    await func(wrappedInfra, payload);
+  } catch (error) {
+    processServiceError(error, logger, {
+      operationId,
+      logPrefix,
+    });
+  }
+};
+
+/** @type ServiceFuncs['wrapInfra'] */
+const wrapInfra = (infra, operationId) => {
+  const { bus, ...rest } = infra;
+  return {
+    ...rest,
+    bus: {
+      command: (commandName, payload) => {
+        const wrappedMeta = { ...payload.meta, operationId };
+        return bus.command(commandName, { ...payload, meta: wrappedMeta });
+      },
+      publish: (eventName, payload) => {
+        const wrappedMeta = { ...payload.meta, operationId };
+        return bus.publish(eventName, { ...payload, meta: wrappedMeta });
+      },
+    },
+  };
+};
```

Comment:
Here is optional metadata injection withing bus using Proxy and partial application function
https://github.com/georgolden/node-soa/blob/wip-fucntional-style/src/infra/nodeBus.js
I use this during domain function creation - in your case it is a name of a command handler or event handler
https://github.com/georgolden/node-soa/blob/wip-fucntional-style/src/dfs/domainFunction.js